pa11ycrawler/cmdline.py

```diff
@@ -0,0 +1,204 @@
+
+from pa11ycrawler.reporter import HtmlReporter
+from pa11ycrawler.settings_helpers import (
+    is_pa11y_setting,
+    is_pa11ycrawler_setting,
+    get_default_settings,
+)
+from pa11ycrawler.spiders.pa11y_spider import Pa11ySpider
+
+# ------------ Main CLI ------------
+
+
+def cli():
+    """
+    Returns the main CLI parser.
+    """
+    parser = argparse.ArgumentParser(prog='pa11ycrawler')
+    subparsers = parser.add_subparsers(title='commands')
+    run_cli(subparsers)
+    json_to_html_cli(subparsers)
+    return parser
+
+# ------------ "run" CLI -----------
+
+
+def run_cli(subparsers):
+    """
+    Creates the parser for the `run` command
+    """
+    run_usage = dedent('''pa11ycrawler run [options] url [url ...]
+
+    ---------------------------------------------------------------
+    NOTE: This only documents a haldful of the possible options.
+    More documentation can be found for the following categories:
+
+    CUSTOM SCRAPY SETTINGS:
+    ** Implemented for pa11ycrawler. (e.g. Not found in Scrapy Docs.)
+    ** All options documented below.
+    ** Most likely to need to be changed.
+
+    PA11Y SETTINGS:
+    ** All options documented below. More details can be found at
+       https://github.com/springernature/pa11y#command-line-interface
+
+    STANDARD SCRAPY SETTINGS:
+    ** http://doc.scrapy.org/en/latest/topics/settings.html
+    ** These can be set using typical flag formatting. For example,
+       --depth-limit=2 will set the DEPTH_LIMIT setting to 2.
+    ---------------------------------------------------------------
+    ''')
+
+    parser_run = subparsers.add_parser(
+        'run',
+        usage=run_usage,
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        help="Run the crawler, checking accessibility along the way!"
+    )
+    parser_run.set_defaults(func=run)
+    parser_run.add_argument(
+        'PA11YCRAWLER_START_URLS',
+        metavar='url',
+        nargs='+',
+        help='URL(s) to crawl'
+    )
+
+    custom_group = parser_run.add_argument_group('custom crawler options')
+    pa11y_group = parser_run.add_argument_group('pa11y options')
+    scrapy_group = parser_run.add_argument_group('scrapy options')
+
+    settings = get_default_settings()
+    keys = settings.attributes.keys()
+    keys.sort()
+
+    for key in keys:
+        if key == 'PA11YCRAWLER_START_URLS':
+            continue
+
+        flag = "--" + key.lower().replace('_', '-')
+        flag_args = get_flag_args(settings.attributes[key].value)
+
+        if is_pa11ycrawler_setting(key):
+            custom_group.add_argument(flag, dest=key, **flag_args)
+        elif is_pa11y_setting(key):
+            pa11y_group.add_argument(flag, dest=key, **flag_args)
+        else:
+            flag_args['help'] = argparse.SUPPRESS
+            scrapy_group.add_argument(flag, dest=key, **flag_args)
+
+
+def run(settings):
+    """
+    Runs the pa11ycrawler.
+    """
+    print 'Running pa11ycrawler...'
```

Comment:
This will log info as it's crawling, so there are progress indicators when the command is running.